lib/em-hiredis/pubsub_client.rb

```diff
@@ -219,14 +219,17 @@ def subscribe_impl(type, subscriptions, channel, cb)
       if subscriptions.include?(channel)
         # Short circuit issuing the command if we're already subscribed
         subscriptions[channel] << cb
-      elsif @connection_manager.state == :failed
-        raise('Redis connection in failed state')
-      elsif @connection_manager.state == :connected
-        @connection_manager.connection.send_command(type, channel)
-        subscriptions[channel] << cb
       else
-        # We will issue subscription command when we connect
-        subscriptions[channel] << cb
+        subscriptions[channel] = []
+        if @connection_manager.state == :failed
+          raise('Redis connection in failed state')
```

Comment:
If we raise, do we really want to add the subscription in line 223? Because if someone reconnects the failed client then we will subscribe, but we've thrown away their callback.